### ISSUE TITLE
RequestDeviceAuthorizationAsync fails when both ClientCredentialStyle not set & no Parameters are set

### ISSUE THREAD

**Issue by lemonlion** [DuendeSoftware/Support#201]
**Which version of Duende IdentityServer are you using?**
IdentityModel 6.0.0

**Which version of .NET are you using?**
.Net 6

**Describe the bug**
When calling RequestDeviceAuthorizationAsync with just Address and ClientId ([as per the example](https://docs.duendesoftware.com/identityserver/v5/reference/endpoints/device_authorization/)) the request fails with a response of "invalid_request".  If you set the ClientCredentialStyle to AuthorizationHeader, or add any parameters to the Parameters property then the request succeeds.

When debugging through the code, the request appears to be failing [on this line in the DeviceAuthorizationEndpoint class](https://github.com/DuendeSoftware/IdentityServer/blob/f00ce9aa732894698edd4d1a09ebbe47ae990f6e/src/IdentityServer/Endpoints/DeviceAuthorizationEndpoint.cs#L66).  This would appear to be caused by the fact that no content header is added, which in turn is because a new FormUrlEncodedContent [is only created if the parameters collection is not empty](https://github.com/IdentityModel/IdentityModel/blob/2e94543bb365995fa3fc9201c407acb2b9578d61/src/Client/Messages/ProtocolRequest.cs#L195) (which is the case for ClientCredentialStyle.PostBody, but not for ClientCredentialStyle.AuthorizationHeader).

**To Reproduce**

1) Call RequestDeviceAuthorizationAsync with valid address, clientId (& clientSecret if necessary)

```
var startAuthorisationResponse = await client.RequestDeviceAuthorizationAsync(new DeviceAuthorizationRequest
{
    Address = disco.DeviceAuthorizationEndpoint,
    ClientId = ClientId,
});
```
2) Note the `{"error":"invalid_request"}` returned.

**Expected behavior**

A valid response

**Log output/exception with stacktrace**

```
[23:00:10 Information] Duende.IdentityServer.Hosting.IdentityServerMiddleware
Invoking IdentityServer endpoint: Duende.IdentityServer.Endpoints.DeviceAuthorizationEndpoint for /connect/deviceauthorization

[23:00:10 Warning] Duende.IdentityServer.Endpoints.DeviceAuthorizationEndpoint
Invalid HTTP request for device authorize endpoint

[23:00:10 Error] Duende.IdentityServer.Endpoints.DeviceAuthorizationEndpoint
Device authorization error: invalid_request:invalid_request

[23:00:10 Information] Serilog.AspNetCore.RequestLoggingMiddleware
HTTP POST /connect/deviceauthorization responded 400 in 8.5777 ms
```

**Additional context**

In practise, to get around the problem, we have just changed the ClientCredentialStyle to PostBody.  Is there any particular disadvantage/benefit of using one ClientCredentialStyle over the other?

**Comment by brockallen** [comment]
This sounds more like an IdentityModel issue, than an IdentityServer one, yes?

**Comment by lemonlion** [comment]
I suspected the mistake was in the IdentityModel code, but as the failure is happening in the IdentityServer code itself, I thought I would check - but it sounds like you're saying the IdentityServer code is behaving as expected?

**Comment by brockallen** [comment]
Oh I see -- when there is no need to pass form params, the IdentityServer check is preventing all requests. Hmm, I'll have a think about this. Thanks.

**Comment by brockallen** [comment]
Hmm... does HTTP Basic auth allow for an empty password? IOW, is this client really authenticating? If not, then the device flow says client_id is required in the post body.

**Comment by lemonlion** [comment]
Are you implying that the problem is most likely with Identity Model for not putting the client_id in the post body when it gets serialized in to a http request?

**Comment by brockallen** [comment]
There's an option for how to send the client id and (optional) secret.

**Comment by lemonlion** [comment]
I'm not sure I understand you.  Are you saying there is an option to set ClientCredentialStyle to ClientCredentialStyle.PostBody which will put the client id in the request?  Yes, I understand that this is the workaround (I said so in the 'Additional context' secition above), but it is not the default option of the ClientCredentialStyle property and [the documentation](https://docs.duendesoftware.com/identityserver/v6/reference/endpoints/device_authorization/) gives an example without setting the ClientCredential style, so the example in the documentation will never work.  So there is a problem somewhere.  Either:

1) The example in the documentation needs to change.  This is the easiest fix, although arguably the implementation of this feature is not intuitive so could do with setting better default values.
**OR**
2) The default value of ClientCredentialStyle needs to change to ClientCredentialStyle.PostBody
**OR**
3) The content header needs to automatically populate even for ClientCredentialStyle.AuthorizationHeader, even when the parameters collection is empty.
**OR**
4) Some logic needs to change in Identity Server validation.

The first option is just a Duende documentation change (I'm not sure where I should report that).  The second and third option are Identity Model code changes as far as I can tell.  The fourth option is an Identity Server change.

Please clarify which option you think should be done (I'm guessing you're thinking number 1 or number 2) and then I'll know where to move forward with this.

**Comment by leastprivilege** [comment]
As this spans IDS and IDM we will discuss.

**Comment by leastprivilege** [comment]
OK - this is fixed in IdentityModel and will be released with 6.1.

Thanks for reporting.
